tools/build_defs/cc_toolchain_util.bzl

```diff
@@ -239,6 +244,15 @@ def get_env_vars(ctx):
                 user_compile_flags = copts,
             ),
         ))
+
+    if cc_toolchain:
+        cc_executable = cc_toolchain.compiler_executable
```

Comment:
compiler_executable is deprecated and I've definitely had this return the wrong value in the case that there are different tools configured in the toolchain dependent upon the features set.
I think the correct thing to use here is `cc_common.get_tool_for_action`

Reply:
Great to know! Thanks for pointing that out 😄 

Updated